lua/refactoring/config.lua

```diff
@@ -1,92 +1,89 @@
-                        type(opts.body) == "table"
-                                and table.concat(opts.body, "\n")
-                            or opts.body,
-                        opts.ret
-                    ),
-                    -- TODO: OBVI THIS NEEDS TO BE DIFFERENT...
-                    call = string.format(
-                        "const %s = %s(%s)",
-                        opts.ret,
-                        opts.name,
-                        table.concat(opts.args, ", ")
-                    ),
-                }
-            end,
-        },
-        lua = {
-            extract_function = function(opts)
-                return {
-                    create = string.format(
-                        [[
+]], opts.name, table.concat(opts.args, ', '), type(opts.body) == 'table' and
+                                           table.concat(opts.body, '\n') or
+                                           opts.body, opts.ret),
+                -- TODO: OBVI THIS NEEDS TO BE DIFFERENT...
+                call = string.format('const %s = %s(%s)', opts.ret, opts.name,
+                                     table.concat(opts.args, ', '))
+            }
+        end
+    },
+    lua = {
+        extract_function = function(opts)
+            return {
+                create = string.format([[
 local function %s(%s)
     %s
     return %s
 end
 
-]],
-                        opts.name,
-                        table.concat(opts.args, ", "),
-                        type(opts.body) == "table"
-                                and table.concat(opts.body, "\n")
-                            or opts.body,
-                        opts.ret
-                    ),
+]], opts.name, table.concat(opts.args, ', '), type(opts.body) == 'table' and
+                                           table.concat(opts.body, '\n') or
+                                           opts.body, opts.ret),
 
-                    call = string.format(
-                        "local %s = %s(%s)",
-                        opts.ret,
-                        opts.name,
-                        table.concat(opts.args, ", ")
-                    ),
-                }
-            end,
-        },
-    },
-    _automation = {},
+                call = string.format('local %s = %s(%s)', opts.ret, opts.name,
+                                     table.concat(opts.args, ', '))
+            }
+        end
+    }
 }
 
-local M = {}
+-- TODO: add a way to accept other formatting / code_gen opts
+function Config:new()
+    local conf = {}
+    setmetatable(conf, Config)
+    local automation = {}
+    automation.current_input_index = 0
+    automation.inputs = {}
+    conf._automation = automation
+    conf.formatting = default_formatting
+    conf.code_generation = default_code_generation
+    return conf
+end
 
-function M.get_config()
-    return config
+function Config:automate_input(inputs)
+    if type(inputs) ~= "table" then inputs = {inputs} end
+    self._reset_inputs(inputs)
```

Comment:
I believe you will lose `self` here.

Make sure you use `self:_reset_inputs(inputs)`